javageci-core/src/main/java/javax0/geci/equals/Equals.java

```diff
@@ -88,11 +89,18 @@ public void postprocess(Source source, Class<?> klass, CompoundParams global) th
     private void generateEqualsHeader(Segment segment, Class<?> klass, CompoundParams global) {
         var equalsMethod = getMethodOrNull(klass, "equals", Object.class);
         var subclassingAllowed = global.is("subclass", config.subclass);
+        var usingSuper = global.is("useSuper", config.useSuper);
         generateEquals = equalsMethod == null || GeciAnnotationTools.isGenerated(equalsMethod);
         writeGenerated(segment, config.generatedAnnotation);
         segment.write("@Override")
                 .write_r("public %sboolean equals(Object o) {", subclassingAllowed ? "final " : "")
                 .write("if (this == o) return true;");
+        try {
+            if(usingSuper && !GeciReflectionTools.getMethod(klass.getSuperclass(), "equals", Object.class).getDeclaringClass().equals(Object.class)) {
```

Comment:
This is a very good solution. I would have stepped through the superclasses till the Object class. It is much shorter and better.